src/util/range.h

```diff
@@ -341,6 +341,18 @@ struct ranget final
     return begin_value == end_value;
   }
 
+  /// Drop the first \p count elements.
+  /// If the range has fewer elements, drops everything.
+  void drop(std::size_t count)
+  {
+    std::size_t i = 0;
+    while(i < count && begin_value != end_value)
```

Comment:
Nit pick, but wouldn't this make for a very nice `for` loop? As is, it looks a bit like assembler programming.